src/quoteoftheday/routes.py

```diff
@@ -1,57 +1,49 @@
 import random
 
 from featuremanagement.azuremonitor import track_event
-from flask import Blueprint, render_template, request, flash, redirect, url_for
+from flask import Blueprint, render_template, request, flash, redirect, url_for, jsonify
 from flask_login import current_user, login_user, logout_user
 from . import azure_app_config, feature_manager, db, bcrypt
 from .model import Quote, Users
 
 bp = Blueprint("pages", __name__)
 
-@bp.route("/", methods=["GET", "POST"])
+@bp.route("/", methods=["GET"])
 def index():
     global azure_app_config
     # Refresh the configuration from App Configuration service.
     azure_app_config.refresh()
-    context = {}
-    user = ""
-    if current_user.is_authenticated:
-        user = current_user.username
-        context["user"] = user
-    else:
-        context["user"] = "Guest"
-    if request.method == "POST":
-        track_event("Liked", user)
-        return redirect(url_for("pages.index"))
+    context = get_user_context()
 
     quotes = [
         Quote("You cannot change what you are, only what you do.", "Philip Pullman"),
     ]
 
-    greeting = feature_manager.get_variant("Greeting", user)
-    show_greeting = ""
+    greeting = feature_manager.get_variant("Greeting", context["user"])
+    greeting_message = ""
     if greeting:
```

Comment:
Shouldn't the greating message come from the variant feature flag regardless its state?